Fixes #29292 - Move pagination options to the client

tbrisker opened the pull request:
<!---

Thank you for contributing to The Foreman, please read the
[following guide](https://www.theforeman.org/contribute.html), in short:

* [Create an issue](https://projects.theforeman.org/projects/foreman/issues)
* Reference the issue via `Fixes #1234` in the commit message
* Prefer present-tense, imperative-style commit messages
* Mark all strings for translation, see [1]
* Suggest prerequisites for testing and testing scenarios following example above.
* Prepend `[WIP]` for work in progress to prevent bots from triggering actions
* Be patient, we will do our best to take a look as soon as we can
* Explain the purpose of the PR, attach screenshots if applicable
* List all prerequisites for testing (e.g. VMware cluster, two smart proxies...)
* Reviewers often use extensive list of items to check, have a look prior submitting [2]
* Be nice and respectful

1: https://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers
2: https://github.com/theforeman/foreman/blob/develop/developer_docs/pr_review.asciidoc
-->

amirfefer committed:
Fixes #29292 - Move pagination options to the client

There is no need to send the pagination options from the server, they
can be generated on the client side.

tbrisker committed:
Refs #29292 - fix review comments

Renamed PaginationHelper to PaginationHooks and removed test that is no
longer needed.

tbrisker commented:
Please squash on merge and see GH-7497 for previous comments.

tbrisker commented:
test failures are all intermittent and unrelated.

amirfefer reviewed (changes requested):
Thanks @tbrisker , tested and works well! 
Though katello needs an update as well - https://github.com/Katello/katello/blob/a046c3b1ce8ff2ef5cc1da6f35b8f0cd02bcbfce/webpack/components/Table/TableWrapper.js#L6

tbrisker commented:
Good find @amirfefer ! Opened https://github.com/Katello/katello/pull/9052 to fix it in katello. Next step might be to move it into the Pagination component itself so consumers don't need to care about it, but let's first get this change in.

tbrisker committed:
Refs #29292 - Remove useless pagination options from audit table

The pagination component already gets the pagination options and doesn't
need them to be passed in from the calling component.

tbrisker commented on webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/components/AuditsTable.js:
```diff
@@ -4,35 +4,30 @@ import { withRenderHandler } from '../../../../common/HOC';
 import AuditsList from '../../../../components/AuditsList';
 import AuditsLoadingPage from './AuditsLoadingPage';
 import Pagination from '../../../../components/Pagination/Pagination';
-import { usePaginationOptions } from '../../../../Root/Context/ForemanContext';
```
Just realized we don't actually need to pass this into the component, as it already defaults to it.

tbrisker commented:
@amirfefer can you take another look? We'll need to merge katello one right after this one. Looks like the reason it is needed in katello is that they use patternfly 4 pagination which expects the options in a bit different manner.

amirfefer reviewed (approved):
Thanks @tbrisker :+1: 
 works well
